lib/determinator/retrieve/http_retriever.rb

```diff
@@ -0,0 +1,27 @@
+require 'faraday'
+
+module Determinator
+  module Retrieve
+
+    class HttpRetriever
+      def initialize(connection:)
+        raise ArgumentError, "client must be a Faraday::Connection" unless connection.is_a?(Faraday::Connection)
+        @connection = connection
+      end
+
+      def retrieve(name)
+        begin
+          response = @connection.get("/features/#{name}")
+          return Determinator::Serializers::JSON.load(response.body) if response.status == 200
+        rescue => e
+          Determinator.notice_error(e)
+        end
+        nil
+      end
+
+      def get_name(url)
+        (url.match('features\/(.*)\z') || [])[1]
```

Comment:
I think this might be better as a class method. It'd be nice to have some YARDoc to explain what it's for, and to point out that it's rather dumb (ie. won't deal with strings like "actor-tracking.example.com/features/some_feature\nsome.other.url.example.com", or "some-invalid-url.example.com/see-theres-no-name-at-the-end/features/")

Reply:
I've added some docs. 

I didn't go to the class method, it just makes usage a little more clumsy in client code.